data-pipeline/src/trace_exporter.rs

```diff
@@ -249,11 +280,21 @@ impl TraceExporter {
                             Ok(body) => Ok(String::from_utf8_lossy(&body).to_string()),
                             Err(err) => {
                                 error!("Error reading agent response body: {err}");
+                                if let Some(ref mut flusher) = self.dogstatsd {
+                                    flusher.send(vec![DogStatsDAction::Count(String::from("datadog.libdatadog.send.errors"),
+                                                                             1,
+                                                                             vec![])]);
+                                }
                                 Ok(String::from("{}"))
                             }
                         },
                         Err(err) => {
                             error!("Error sending traces: {err}");
+                            if let Some(ref mut flusher) = self.dogstatsd {
+                                flusher.send(vec![DogStatsDAction::Count(String::from("datadog.libdatadog.send.errors"),
+                                                                         1,
+                                                                         vec![])]);
```

Comment:
`Vec::default()` is probably preferable here? I believe vec![] is going to allocate memory here, whereas Vec::new() will not allocate memory until elements get added to it. Plus, it more clearly shows your intention of passing an empty vec.